services/github/github-constellation.js

```diff
@@ -70,8 +80,21 @@ class GithubConstellation {
       log.error(e)
     }
 
+    // Reserve a subset of scoped tokens from the total set
+    // to be used for queries which require an explicit scope,
+    // while leaving a sufficient amount of tokens (scoped or unscoped)
+    // for the bulk of our requests which don't care about scopes.
+    this._maxNumReservedScopedTokens = Math.floor(tokens.length * 0.15)
     tokens.forEach(tokenString => {
-      this.apiProvider.addToken(tokenString)
+      const [token, scopes] = tokenString.split(persistenceScopeDelimiter)
+      this._tokenScopes[token] = scopes || null
```

Comment:
i opted to make this explicitly `null` vs the existing `undefined` as I found that made troubleshooting and debugging easier. kept it in here as i don't think null vs. undefined as any runtime impact given the code, since i can see it being beneficial for any future debugging needs